### PR TITLE
doc/tut1: new section on functions and methods 

### PR DESCRIPTION
**Pull request by ZoomRmc**
This PR adds a new section on functions and methods to the first part of the tutorial.

The added text clarifies the differences between the three and provides some benefits of using `func`s which explains their existence in the language.

This PR aims to answer a repeating question of "which of the three should I use?" coming from new language users, which was observed on multiple occasions by the @PMunch and others.

Most of the work is by @PMunch, part about functions coauthored by me in a declined edit of the merged PR (https://github.com/nim-lang/Nim/pull/19207)

### CODE REVIEW

**Review by saem — CHANGES_REQUESTED** [review]
Change all references to nimskull

**Comment by alaviss** [comment]
You should change the commit author to `PMunch <peterme@peterme.net>`, then add the following `Co-authored-by` as evident by [upstream commit](https://github.com/nim-lang/Nim/commit/c658de24b04be9f898359a4091c5815f28de8785):

```
Co-authored-by: Danil Yarantsev <tiberiumk12@gmail.com>
Co-authored-by: konsumlamm <44230978+konsumlamm@users.noreply.github.com>
Co-authored-by: Zoom <ZoomRmc@users.noreply.github.com>
```

When backporting I'd prefer that we keep as much authoring metadata as possible.

**Comment by saem** [comment]
This is a good idea, in starting to review further.

We have pr guidelines so could you start sprucing up the description accordingly for others? I don't want to have strong references into Nim as we do things differently and I don't want to give people the wrong omissions as they try and soft through what does and doesn't make sense.

**Comment by ZoomRmc** [comment]
> Change all references to nimskull

Wait, do you mean the whole doc? Changing only in the edited section doesn't make a lot of sense.

> could you start sprucing up the description accordingly for others

Not sure I'm following. Sprucing up what description accordingly to what?

**Comment by saem** [comment]
> > Change all references to nimskull
> 
> Wait, do you mean the whole doc? Changing only in the edited section doesn't make a lot of sense.

Let me clarify, I recently added `|NimSkull|` and `|Nim|` (not preferred) to `rstcommon.rst`, you can check them out [here](https://github.com/nim-works/nimskull/blob/devel/doc/rstcommon.rst#L21). Please use those at least for the parts you edited -- broader change can be separate, but I'd prefer that first to be honest.

> > could you start sprucing up the description accordingly for others
> 
> Not sure I'm following. Sprucing up what description accordingly to what?

We have a PR template, that should have popped up, please see the contents [here](https://raw.githubusercontent.com/nim-works/nimskull/devel/.github/pull_request_template.md), thanks.

Specifically, here are some good starting points:
* shortening the title, no reference to upstream -- they are not
* `doc/tutorial:` is likely a reasonable prefix
* the PR template tips around current state and why these are good changes

Hope those clarify things, keep the questions coming we'll help ya along!

File: doc/tut1.rst [hunk 1]
```diff
@@ -595,8 +595,10 @@ Procedures
 
 To define new commands like `echo <system.html#echo,varargs[typed,]>`_
 and `readLine <io.html#readLine,File>`_ in the examples, the concept of a
-*procedure* is needed. (Some languages call them *methods* or *functions*.)
-In Nim new procedures are defined with the `proc` keyword:
+*procedure* is needed. You might be used to them being called *methods* or
+*functions* in other languages, but Nim
```
Comment (saem) [inline doc/tut1.rst]: This and the next line have Nim

**Review by saem — CHANGES_REQUESTED** [review]
Sneaky ones

File: doc/tut1.rst (hunk 1 above)
Comment (ZoomRmc) [reply doc/tut1.rst]: Didn't consider that paragraph as reworked :)

Comment (saem) [reply doc/tut1.rst]: I was playing by smelt it dealt it rules. 😁

OK, it can live this PR, but its day will come.

**Review by saem — APPROVED** [review]
Bors r+

**Comment by saem** [comment]
Thanks, @ZoomRmc!

Happy early first commit into nimskull. 🎉 👍🏽 😄

**Pull request merged**